From 30ba442171596f1c1a121dec6101ed320c026d61 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 14 Jan 2011 15:13:53 +0000
Subject: [PATCH 102/117] RHEL 6: Require external hex editor set with
 $HEXEDITOR.

RHEL 6 doesn't have the hexedit program or any suitable replacement.
Require that the user sets $HEXEDITOR to make this command work.
(cherry picked from commit 1ee31426e13446bc5fcce242eb496a01d8a78d8c)
---
 fish/guestfish.pod   | 3 +--
 fish/hexedit.c       | 6 ++++--
 generator/actions.ml | 9 +++++----
 3 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/fish/guestfish.pod b/fish/guestfish.pod
index 6bfcc4e..70c4d4b 100644
--- a/fish/guestfish.pod
+++ b/fish/guestfish.pod
@@ -1265,8 +1265,7 @@ Set the command prompt.  See L</PROMPT>.
 =item HEXEDITOR
 
 The L</hexedit> command uses C<$HEXEDITOR> as the external hex
-editor.  If not specified, the external L<hexedit(1)> program
-is used.
+editor.  If not set, then an error message is printed.
 
 =item HOME
 
diff --git a/fish/hexedit.c b/fish/hexedit.c
index af72ebb..c46b747 100644
--- a/fish/hexedit.c
+++ b/fish/hexedit.c
@@ -116,8 +116,10 @@ run_hexedit (const char *cmd, size_t argc, char *argv[])
 
   /* Choose an editor. */
   editor = getenv ("HEXEDITOR");
-  if (editor == NULL)
-    editor = "hexedit";
+  if (editor == NULL) {
+    fprintf (stderr, _("the $HEXEDITOR environment variable must be set\n"));
+    return -1;
+  }
 
   snprintf (buf, sizeof buf, "/dev/fd/%d", fd);
 
diff --git a/generator/actions.ml b/generator/actions.ml
index f0c157d..5f1bd40 100644
--- a/generator/actions.ml
+++ b/generator/actions.ml
@@ -11008,7 +11008,7 @@ See L</WILDCARDS AND GLOBBING>." };
  hexedit <filename|device> <max>
  hexedit <filename|device> <start> <max>
 
-Use hexedit (a hex editor) to edit all or part of a binary file
+Use a hex editor to edit all or part of a binary file
 or block device.
 
 This command works by downloading potentially the whole file or
@@ -11032,9 +11032,10 @@ To edit the superblock of an ext2 filesystem on C</dev/sda1>, do:
 
 (assuming the superblock is in the standard location).
 
-This command requires the external L<hexedit(1)> program.  You
-can specify another program to use by setting the C<HEXEDITOR>
-environment variable.
+This command requires an external hexeditor program.  You
+can specify the program to use by setting the C<HEXEDITOR>
+environment variable.  If C<HEXEDITOR> is not set then this
+prints an error.
 
 See also L</hexdump>." };
 
-- 
2.5.0

